### ISSUE TITLE
upgrade-perl requires reinstalling all user-installed XS modules

### ISSUE THREAD

**Issue by grr** [gugod/App-perlbrew#680]
Whenever I use `upgrade-perl` to upgrade to a point release (ex. 5.30.0 to 5.30.1) all user-installed (non-core) XS modules can not be found. The docs say:

>Minor Perl releases (ex. 5.x.*) are binary compatible with one another, so this command offers you the ability to upgrade older perlbrew environments in place.

which implies that `upgrade-per`l that no additional steps need to be taken. Either this is expected behavior and the docs need updating, or this is a bug or configuration issue which needs addressing.

My `@INC`:
    /Users/grr/src/perlbrew/perls/stable/lib/site_perl/5.30.1/darwin-thread-multi-2level
    /Users/grr/src/perlbrew/perls/stable/lib/site_perl/5.30.1
    /Users/grr/src/perlbrew/perls/stable/lib/5.30.1/darwin-thread-multi-2level
    /Users/grr/src/perlbrew/perls/stable/lib/5.30.1
    /Users/grr/src/perlbrew/perls/stable/lib/site_perl/5.30.0
    /Users/grr/src/perlbrew/perls/stable/lib/site_perl

**Comment by erickpaquin** [comment]
Hi @grr . Is this still an issue? If so what version of perlbrew are you using? Thanks.

**Comment by grr** [comment]
> Hi [@grr](https://github.com/grr) . Is this still an issue? If so what version of perlbrew are you using? Thanks.

Yes, just upgraded to Perl 5.40.1 with perlbrew 1.01 and same thing happened. Doing `perlbrew clone-modules 5.40` or `perlbrew list-modules | cpanm` will fix it, but sometimes module installation will fail because the upgrade broke only the XS part of modules and the installation of the new modules thinks it's prereqs are already installed and fails on tests. Running `cpanm -nf` avoids the errors by disabling tests and forcing re-installation.
